### PR TITLE
added support for Haiku

### PR DESCRIPTION
**Pull request by kenmays**
Haiku - https://www.haiku-os.org
Patch contributor: Gerasim Troeglazov <3dEyes@gmail.com>
Ref: https://github.com/haikuports/haikuports/tree/master/media-video/qmplay2/patches

### CODE REVIEW

File: src/modules/MediaKit/RingBuffer.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "RingBuffer.hpp"
+
+RingBuffer::RingBuffer( int size )
+{
+	 initialized = false;
```
Comment (zaps166) [inline src/modules/MediaKit/RingBuffer.cpp:9]: Is it possible to change all indentations to spaces?

**Review by zaps166 — COMMENTED** [review]
Thanks for the PR!

File: src/qmplay2/QMPlay2Core.cpp [hunk 2]
```diff
@@ -491,6 +491,11 @@ QStringList QMPlay2CoreClass::getModules(const QString &type, int typeLen) const
 #elif defined Q_OS_WIN
     if (type == "videoWriters")
         defaultModules << "OpenGL 2" << "DirectDraw";
+#elif defined Q_OS_HAIKU
+	if ( type == "videoWriters" )
+		defaultModules << "QPainter";
```
[the remark this answers is not in the document]
Comment (kenmays) [reply src/qmplay2/QMPlay2Core.cpp:496]: Yes - Using Mesa. Not HW accelerated yet. I'm currently running QMPlay2 w/ OpenGL enabled.

[the remark this answers is not in the document]
Comment (zaps166) [reply src/qmplay2/QMPlay2Core.cpp:496]: Ok, so it's better to keep `QPainter` as default due to performance reason.

File: src/gui/Main.cpp [hunk 3]
```diff
@@ -918,5 +922,9 @@ int main(int argc, char *argv[])
     if (canDeleteApp)
 #endif
         delete qApp;
+
+#ifdef Q_OS_HAIKU
+	kill(::getpid(), SIGKILL);
```
[the remark this answers is not in the document]
Comment (kenmays) [reply src/gui/Main.cpp:927]: Can use SIGTERM.

File: src/qmplay2/YouTubeDL.cpp [hunk 4]
```diff
@@ -37,9 +37,13 @@ static QMutex g_mutex(QMutex::Recursive);
 
 QString YouTubeDL::getFilePath()
 {
+#ifdef Q_OS_HAIKU
+    return "/bin/yt-dlp"
```
[the remark this answers is not in the document]
Comment (kenmays) [reply src/qmplay2/YouTubeDL.cpp:41]: Haiku uses yt-dlp (not natively, additional software). Seems a reliable method for now.

File: src/gui/Main.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (zaps166) [reply src/gui/Main.cpp:927]: No, my question is why is it needed? Does Haiku needs it in all applications? Or it's a workaround for crash at exit?

File: src/modules/MediaKit/RingBuffer.cpp (hunk 1 above)
Comment (kenmays) [reply src/modules/MediaKit/RingBuffer.cpp:9]: Yes. Feel free to change to fit your coding style.

File: src/gui/Main.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (kenmays) [reply src/gui/Main.cpp:927]: For SIGKILL in all apps, no. Please, use what is best for app.

File: src/modules/MediaKit/SndPlayer.hpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
+#ifndef PULSE_HPP
```
[the remark this answers is not in the document]
Comment (kenmays) [reply src/modules/MediaKit/SndPlayer.hpp:1]: Y

File: src/modules/MediaKit/RingBuffer.hpp [hunk 6]
```diff
@@ -0,0 +1,31 @@
+#ifndef __RING_BUFFER_H__
```
[the remark this answers is not in the document]
Comment (kenmays) [reply src/modules/MediaKit/RingBuffer.hpp:1]: Y

**Comment by kenmays** [comment]
Some standards issue with 'pragma once' usage on Net: "The fact of the matter is, #ifndef is standardized with standard behavior whereas #pragma once is not, and #ifndef also handles weird filesystem and search path corner cases whereas #pragma once can get very confused by certain things, leading to incorrect behavior which the programmer has no control over."

**Comment by zaps166** [comment]
I'll look at this and merge, thanks!

**Comment by zaps166** [comment]
I prefer `#pragma once` - short (1 line), no mistakes such as wrong file name or different names in `ifndef` and `define` because of typo (look here: http://libcddb.sourceforge.net/API/cddb__log_8h-source.html).

**Comment by zaps166** [comment]
@kenmays 
- Can you check if `rubberband` is working on Haiku?
- Can you tell me if it's possible to use Vulkan (software, lavapipe) on Haiku?

**Comment by kenmays** [comment]
1. Yes. Rubberband is available on Haiku.
2. Yes. Vulkan (Mesa, lavapipe) is working on Haiku.

**Comment by zaps166** [comment]
@kenmays Thanks, merged to `master`.
Please test on Haiku and create a PR if it's not working (including rubberband)! Could you check Vulkan swrast now?

**Pull request closed without merging**